dev/tasks/crossbow.py

```diff
@@ -357,81 +360,77 @@ def email(self):
 
 
 @click.group()
-def crossbow():
-    pass
-
-
-def github_token_validation_callback(ctx, param, value):
-    if value is None:
+@click.option('--github-token', '-t', default=None,
+              help='OAuth token for Github authentication')
+@click.option('--arrow-path', '-a',
+              type=click.Path(exists=True), default=DEFAULT_ARROW_PATH,
+              help='Arrow\'s repository path. Defaults to the repository of '
+                   'this script')
+@click.option('--queue-path', '-q',
+              type=click.Path(exists=True), default=DEFAULT_QUEUE_PATH,
+              help='The repository path used for scheduling the tasks. '
+                   'Defaults to crossbow directory placed next to arrow')
+@click.pass_context
+def crossbow(ctx, github_token, arrow_path, queue_path):
+    if github_token is None:
         raise click.ClickException(
             'Could not determine GitHub token. Please set the '
             'CROSSBOW_GITHUB_TOKEN environment variable to a '
             'valid github access token or pass one to --github-token.'
         )
-    return value
 
+    ctx.obj['arrow'] = Repo(Path(arrow_path))
+    ctx.obj['queue'] = Queue(Path(queue_path), github_token=github_token)
 
-github_token = click.option(
-    '--github-token',
-    default=None,
-    envvar='CROSSBOW_GITHUB_TOKEN',
-    help='OAuth token for Github authentication',
-    callback=github_token_validation_callback,
-)
 
-
-def config_path_validation_callback(ctx, param, value):
-    with Path(value).open() as fp:
+def load_tasks_from_config(config_path, task_names, group_names):
+    with Path(config_path).open() as fp:
         config = yaml.load(fp)
-    task_names = ctx.params['task_names']
+
+    valid_groups = set(config['groups'].keys())
```

Comment:
Because we have more tasks now, I've setup [groups](https://github.com/kszucs/arrow/blob/b299c6ff515a33cffdabd0b974c12cdffcb97ebc/dev/tasks/tasks.yml#L18) which can contain arbitrary task names.
A more convenient method to submit related tasks `python crossbow.py submit -g conda -g wheel`